### PR TITLE
boards/serpente: add serpente board

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

The [Serpente boards](https://serpente.solder.party/) are low-cost development boards designed to be used with Adafruit’s CircuitPython.

Three versions of the board board exist, but they all the same except for the USB connector.
The standard Serpente board contains a female USB Type-C connector, the Serpente Plug C
contains a male USB Type-C connector, and the Serpente Plug uses the board itself as a Type-A USB plug.

Serptene R2 pinout

The board comes with the [UF2 bootloader](https://learn.adafruit.com/adafruit-feather-m0-express-designed-for-circuit-python-circuitpython/uf2-bootloader-details) that works very much like the arduino bootloader from #12304, but uses a different memory address & magic.

### Testing procedure

Use `BOARD=serpente` with the `make` command.<br/>
Example with `micropython` application:
```
     make BOARD=serpente -C examples/micropython flash
```

RIOT will automatically trigger a reset to the bootloader, but this only works if RIOT is still running on the board.
If your application has crashed or got erased, `make flash` will not be able to trigger a bootloader reset.

To manually enter the bootloader, quickly press the Reset button twice.
You know you've successfully entered the bootloader when the LED begins flashing blue.

Sometimes flashing fails and the board gets stuck in the bootloader.
In this case, just run `make flash` again when the device is not busy anymore.

### Issues/PRs references

requires #13652 for proper USB ID.

### CODE REVIEW

**Comment by fjmolinas** [comment]
@benpicco can you provide some test output?

**Comment by benpicco** [comment]
> @benpicco can you provide some test output?

#### tests/periph_uart

I connected UART0 TX with UART1 RX and UART0 RX with UART1 TX

```
2020-04-26 20:26:01,282 # ================================
2020-04-26 20:26:01,284 # This application is intended for testing additional UART
2020-04-26 20:26:01,284 # 
2020-04-26 20:26:01,285 # Available devices:               2
> init 0 115200
2020-04-26 20:26:06,239 #  init 0 115200
2020-04-26 20:26:06,241 # Success: Initialized UART_DEV(0) at BAUD 115200
2020-04-26 20:26:06,491 # UARD_DEV(0): test uart_poweron() and uart_poweroff()  ->  [OK]
> ini1 1152000
2020-04-26 20:26:09,927 #  init 1 115200
2020-04-26 20:26:09,929 # Success: Initialized UART_DEV(1) at BAUD 115200
2020-04-26 20:26:10,179 # UARD_DEV(1): test uart_poweron() and uart_poweroff()  ->  [OK]
> send 0 Hello\ UART1
2020-04-26 20:26:21,703 #  send 0 Hello\ UART1
2020-04-26 20:26:21,704 # UART_DEV(0) TX: Hello UART1
2020-04-26 20:26:21,705 # Success: UART_DEV(1) RX: [Hello UART1]\n
> send 1 Hello\ UART0
2020-04-26 20:26:27,511 #  send 1 Hello\ UART0
2020-04-26 20:26:27,512 # UART_DEV(1) TX: Hello UART0
2020-04-26 20:26:27,514 # Success: UART_DEV(0) RX: [Hello UART0]\n
```

A patch was needed to fix `tests/periph_uart` for the non `STDIO_UART_DEV` case that will be in a separate PR.

File: drivers/mtd_spi_nor/mtd_spi_nor.c [hunk 1]
```diff
@@ -27,10 +27,10 @@
 #include "mtd.h"
 #if MODULE_XTIMER
 #include "xtimer.h"
-#include "timex.h"
 #else
 #include "thread.h"
 #endif
+#include "timex.h"
```
[the remark this answers is not in the document]
Comment (benpicco) [reply drivers/mtd_spi_nor/mtd_spi_nor.c:33]: It's not needed anymore.

File: boards/serpente/Makefile.include [hunk 2]
```diff
@@ -0,0 +1,20 @@
+PORT_LINUX ?= /dev/ttyACM0
+PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.usbmodem*)))
```
[the remark this answers is not in the document]
Comment (benpicco) [reply boards/serpente/Makefile.include:2]: Dropped it.

**Comment by benpicco** [comment]
I rebased because the MTD API now requires additional parmaters

File: boards/serpente/Makefile.include (hunk 2 above)
[the remark this answers is not in the document]
Comment (benpicco) [reply boards/serpente/Makefile.include]: Hmm

> `/home/benpicco/dev/RIOT/tests/pkg_littlefs2/../../Makefile.include:703: *** No port set.  Stop.`

[the remark this answers is not in the document]
Comment (fjmolinas) [reply boards/serpente/Makefile.include:2]: I think I was looking at the wrong file `makefiles/boards/sam0.inc.mk` sorry.

[the remark this answers is not in the document]
Comment (fjmolinas) [reply boards/serpente/Makefile.include:2]: As before is fine, sorry for the noise.

**Comment by fjmolinas** [comment]
Only the doxygen image issue remains, I tried a couple of things but was not succesfull..

**Comment by benpicco** [comment]
This should fix it.

**Comment by fjmolinas** [comment]
Looks good now, please squash!

**Comment by benpicco** [comment]
squashed.

**Comment by benpicco** [comment]
Thank you for the review!

**Pull request merged**